### PR TITLE
Medkits and Auto-equip

### PR DESCRIPTION
**Pull request by Rohesie**
- Medkits are large and don't fit in backpacks anymore, to avoid the magical multiplication of storage space. In compensation they provide much more useful contents.
- Doctors start with a surgical mask, a medkit pouch instead of the useless syringe pouch, and a medbelt which contains a stasis bag, roller bed and defib. They should be ready to heal, minimizing prep time.
- MTs start with welding goggles and a welding pack in their hand instead of the useless scanner.
- Auto-equip functions implemented. Press E to put items inside an open container or inside a pouch. Like the medkit one, for example.
- Knives can be put in boots with E, and ties (webbing/vest) in jumpsuits.

### CODE REVIEW

**Comment by Surrealaser** [comment]
@Rohesie Regular backpacks should be able to fit medikits; might make them worth at least half a shit.

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -479,6 +479,24 @@ cases. Override_icon_state should be a list.*/
 					if(S.can_be_inserted(src))
 						return TRUE
 				return FALSE
+			if(EQUIP_IN_STORAGE)
+				if(H.s_active)
+					var/obj/item/storage/S = H.s_active
+					if(S.can_be_inserted(src))
+						return TRUE
+				return FALSE
```
Comment (Qustinnus) [inline code/game/objects/items.dm]: `if (!H.s_active)
    return FALSE
`

avoid unnecesary nesting

File: code/game/objects/items.dm [hunk 2]
```diff
@@ -479,6 +479,24 @@ cases. Override_icon_state should be a list.*/
...
+			if(EQUIP_IN_L_POUCH)
+				if(H.l_store && istype(H.l_store, /obj/item/storage/pouch))
+					var/obj/item/storage/S = H.l_store
+					if(S.can_be_inserted(src))
+						return TRUE
+				return FALSE
```
Comment (Qustinnus) [inline code/game/objects/items.dm]: ditto

File: code/game/objects/items.dm [hunk 3]
```diff
@@ -479,6 +479,24 @@ cases. Override_icon_state should be a list.*/
...
+			if(EQUIP_IN_R_POUCH)
+				if(H.r_store && istype(H.r_store, /obj/item/storage/pouch))
+					var/obj/item/storage/S = H.r_store
+					if(S.can_be_inserted(src))
+						return TRUE
+				return FALSE
```
Comment (Qustinnus) [inline code/game/objects/items.dm]: ditto

File: code/modules/mob/mob.dm [hunk 4]
```diff
@@ -138,13 +138,16 @@
...
-	if(!istype(W)) return
+/mob/proc/equip_to_slot_if_possible(obj/item/W, slot, ignore_delay = TRUE, del_on_fail = FALSE, disable_warning = FALSE, redraw_mob = TRUE, permanent = FALSE)
+	if(!istype(W))
+		return
 
 	if(!W.mob_can_equip(src, slot, disable_warning))
-		if(del_on_fail) cdel(W)
+		if(del_on_fail)
+			cdel(W)
 		else
```
Comment (Qustinnus) [inline code/modules/mob/mob.dm]: else if

**Comment by Supermichael777** [comment]
Questnnus that's just how TG roles, both are equivalent anyhow

**Comment by Qustinnus** [comment]
they are not equivelant.

**Comment by Rohesie** [comment]
This has been testmerged for a while.

File: code/modules/mob/living/carbon/human/inventory.dm [hunk 5]
```diff
@@ -71,18 +71,26 @@
 			return has_limb("chest")
 		if(WEAR_ACCESSORY)
 			return has_limb("chest")
+		if(EQUIP_IN_BOOT)
+			return has_limb("r_foot") && has_limb("l_foot")
 		if(WEAR_IN_BACK)
-			return TRUE
+			return has_limb("chest")
```
Comment (Qustinnus) [inline code/modules/mob/living/carbon/human/inventory.dm]: im pretty sure you cant remove your chest

Comment (Rohesie) [reply code/modules/mob/living/carbon/human/inventory.dm]: I'm just replicating already existing checks. The purpose I can see is for re-using the proc with modified species that descend from humans (or different if the proc is changed to `living` or `mob`.

Comment (Qustinnus) [reply code/modules/mob/living/carbon/human/inventory.dm]: I suppose, I feel like it's a bit of a waste of cpu use but I suppose there is always modularity to think of.

**Pull request merged**